### DISCUSSION TITLE
Please remove the search quality score or make it transparent

### DISCUSSION THREAD

**Discussion by Thanga-Ganapathy** [npm/feedback#1104] [Website] [answered]
I'm creating this after tired of waiting many years to fix this.

I have some pkgs published, searching through your site with appropriate terms returns invalid results.

The main issue revolves around the `quality` score.

As your [Package search rank criteria](https://docs.npmjs.com/searching-for-and-choosing-packages-to-download#package-search-rank-criteria) says

`Quality includes considerations such as the presence of a README file, stability, tests, up-to-date dependencies, custom website, and code complexity.`

`README file`: Fair enough.

`stability`: Here what is the stability mean?

`tests`: Fair enough but I don't know how you could be able to quantify this, simply having tests doesn't mean the pkg is healthy.

`up-to-date dependencies`: My package does not always want the updated one for various reasons, it is up to the pkg author unless there is a security issues in the dependencies.

`custom website`: Why would a pkg required to have a custom website? some of my packages doesn't require one but the readme file is more than enough.

`code complexity`: This is mostly subjective, I found an issue related to this, our pkg codes are minified, the un-minified plain code pkgs are having better quality score, which is not fair. I think the code minfication is the default build process for the most pkgs using compiler like `esbuild`, `tsup`, `vite`, etc. before publishing for size reduction.

The Problem:

1. First of all, the quality score calculations should be transparent to the public, so that people publishing will be careful and will give a chance to rectify.
2. Even though you have filter in search page like `Sort Packages`, which simply does not work that way, for e.g., if you select `Maintenance` option, then the search results must be appear based on the packages Maintenace, but here the quality value takes precedence and giving misleading results. Try this: https://www.npmjs.com/search?q=terminal%20progress&ranking=maintenance
3. One of my pkg was cloned and published with different scoped by somebody, the search results featured that pkg top of my original pkg, how come this could happen with same code even with better maintenance of mine? Is the `provenance` not considered here?

This is a serious issue; this affects many real value packages visibility.

In a similar discussion here, the maintainer said they are working on this to improve, but passed many years without any announcement or any progress is very much **injustice** to the developer community.

The solution:

Please remove the quality score or make it transparent.

Please fix this as ASAP because this may be your one of issues or something, but this is real issue & dream breaking for our dev community not getting any recognitions.

**Comment by leobalter** [comment, marked as answer]
Thanks! We are aware of this and we are planning to remove [PQM score](https://docs.npmjs.com/searching-for-and-choosing-packages-to-download#package-search-rank-criteria) but I cannot share a timeline at this moment.

**Comment by Thanga-Ganapathy** [comment]
@leobalter Thanks for the reply and I hope it doesn't take again many years to fix.
